### PR TITLE
Wib 973/rejected sellers endpoint [develop]

### PR DESCRIPTION
**Pull request by danielfx90**
### What does this Pull Request do?
It includes the rejected sellers in the notarization results endpoint.

### Why was this Pull Request needed?
Rejected sellers do not get to know in the app. Their activity panel ends up with pending transactions forever.

### Does this Pull Request meet the acceptance criteria?
- [x] I have read the [Contribution guidelines](https://github.com/wibsonorg/notary-sdk/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](https://github.com/wibsonorg/notary-sdk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: notary-api/src/jobs/paymentsHelpers.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    R.groupBy(R.prop('id')), // Object<BatPayId, Array<Pair<BatPayId,Address>>>
+    R.map(R.pluck('address')), // Object<BatPayId, Array<Address>>
+    R.mapObjIndexed(addresses => ({ [result]: addresses })),
+    // Object<BatPayId, Object<result,Array<Address>>>
+  );
+
+  return sellers => (sellers ? reformat(sellers) : {});
+};
+
+const mergeByBatPayId = (id, left, right) => ({ ...left, ...right });
```
Comment (Meketh) [inline notary-api/src/jobs/paymentsHelpers.js]: We can omit the `mapObjIndexed ` if we just.
```suggestion
const mergeByBatPayId = (id, completed, rejected) => ({ completed, rejected });
```
And then we won't need `reformatNotarizations` we can just use `reformat` instead.

File: notary-api/src/jobs/paymentsHelpers.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import R from 'ramda';
+
+const reformatNotarizations = (result) => {
```
Comment (Meketh) [inline notary-api/src/jobs/paymentsHelpers.js]: We don't need a factory here.

File: notary-api/src/jobs/paymentsHelpers.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+const mergeByBatPayId = (id, left, right) => ({ ...left, ...right });
+
+const reformatCompleted = reformatNotarizations('completed');
+const reformatRejected = reformatNotarizations('rejected');
+
+export const getResultsByBatPayId = (completedSellers, rejectedSellers) => {
+  const completed = reformatCompleted(completedSellers);
+  const rejected = reformatRejected(rejectedSellers);
+  return R.mergeWithKey(mergeByBatPayId, completed, rejected);
```
Comment (Meketh) [inline notary-api/src/jobs/paymentsHelpers.js]: At the end this line will be:
```suggestion
  return R.mergeWithKey(
    (id, completed, rejected) => ({ completed, rejected }),
    reformat(completedSellers),
    reformat(rejectedSellers));
```

File: notary-api/src/operations/completeNotarization.js [hunk 4]
```diff
@@ -17,38 +18,47 @@ import { jobify } from '../utils/jobify';
...
+    }) => {
+      const validatedSellers = sellers.map(seller => ({
         ...seller,
         result: getResultFromValidation(validatorResult.find(({ id }) => id === seller.address)),
-      }))
-      .filter(v => v.result !== 'rejected')
-      .map(async (seller) => {
+      }));
+
+      const [rejectedSellers, goodSellers] = R.partition(
```
Comment (Meketh) [inline notary-api/src/operations/completeNotarization.js]: rejected => 😈 
accepted => 👼

File: notary-api/src/operations/completeNotarization.js [hunk 5]
```diff
@@ -17,38 +18,47 @@ import { jobify } from '../utils/jobify';
...
+        R.propEq('result', 'rejected'),
+        validatedSellers,
+      );
+
+      const sellersPaid = await Promise.all(goodSellers.map(async (seller) => {
```
Comment (Meketh) [inline notary-api/src/operations/completeNotarization.js]: No one has been paid yet.

File: notary-api/src/jobs/payments.js [hunk 6]
```diff
@@ -28,24 +30,23 @@ export async function sendUnlock(payIndex, registerPaymentHash) {
...
-  const fee = (price * (sellers.length * (notarizationPercentage / 100))) + notarizationFee;
+  const fee = (price * (paidSellers.length * (notarizationPercentage / 100))) + notarizationFee;
   validateOrThrow(Number(registerPayment.fee) === fee, 'fee did not match requested fee');
 
-  const addressesByBatPayId = sellers.reduce((acc, seller) => (
-    {
-      ...acc, [seller.id]: acc[seller.id] ? [...acc[seller.id], seller.address] : [seller.address],
-    }), {});
-
+  const addressesByBatPayId = getResultsByBatPayId(paidSellers, rejectedSellers);
```
[the remark this answers is not in the document]
Comment (danielfx90) [reply notary-api/src/jobs/payments.js]: `payments.js `seems already too long. I'd rather have this tiny functionality in another file and leave this one to the `sendUnlock` method only.

File: notary-api/package.json [hunk 7]
```diff
@@ -43,6 +43,7 @@
     "morgan": "1.9.1",
     "node-forge": "^0.7.1",
     "openpgp": "3.1.0",
+    "ramda": "0.26.1",
```
[the remark this answers is not in the document]
Comment (danielfx90) [reply notary-api/package.json]: 🕺

File: notary-api/src/operations/completeNotarization.js (hunk 4 above)
Comment (danielfx90) [reply notary-api/src/operations/completeNotarization.js]: ramda 👑 
lodash 🗿

**Pull request merged**